### PR TITLE
Dirty parent when removng a child node.

### PR DESCRIPTION
**Pull request by Ms2ger**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8315)

### CODE REVIEW

**Comment by Ms2ger** [comment]
@bors-servo r=pcwalton

Rebase of #8139

**Comment by eefriedman** [comment]
```
Unexpected Results
==================

/css21_dev/html4/block-in-inline-remove-000.htm
-----------------------------------------------
PASS expected FAIL [Parent]
/css21_dev/html4/block-in-inline-remove-002.htm
-----------------------------------------------
PASS expected FAIL [Parent]
```

On both Mac and Linux.

**Comment by jdm** [comment]
On one hand, I don't recall seeing these failures before:
```

  ▶ FAIL [expected PASS] /_mozilla/css/nth_last_of_type_pseudo_a.html
  └   → /_mozilla/css/nth_last_of_type_pseudo_a.html 4a7970049c8e886dceb8f5a5dfc3479c0b1fa0be
/_mozilla/css/nth_last_of_type_pseudo_b.html f3c5e7f8004718c306b56f0183c3c0f233daf37d
Testing 4a7970049c8e886dceb8f5a5dfc3479c0b1fa0be == f3c5e7f8004718c306b56f0183c3c0f233daf37d

  ▶ TIMEOUT [expected PASS] /_mozilla/css/stacked_layers.html
```
On the other hand I don't think they showed up on previous runs :<

**Comment by asajeffrey** [comment]
I ran those failing tests locally, and they passed :( intermittent failures, oh joy.

**Comment by jdm** [comment]
@asajeffrey I think you need to rebase this again so we can try to merge it.

**Comment by asajeffrey** [comment]
Rebased, and took the opportunity to fix a typo in the commit message.

File: tests/wpt/metadata-css/css21_dev/html4/block-in-inline-remove-000.htm.ini [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [block-in-inline-remove-000.htm]
   type: reftest
-  expected: FAIL
+  expected: PASS
```
Comment (waddlesplash) [inline tests/wpt/metadata-css/css21_dev/html4/block-in-inline-remove-000.htm.ini]: Isn't the norm to just delete the file now, and the testdriver will assume a pass is expected?

**Pull request merged**